ignition/openapi/vim_infrastructure.yaml

```diff
@@ -1,113 +1,190 @@
           content:
             application/json:
               schema:
                 $ref: "#/components/schemas/CreateInfrastructureResponse"
+              examples:
+                response:
+                  $ref: "#/components/examples/ExampleCreateInfrastructureResponse"
+                async-response:
+                  $ref: "#/components/examples/ExampleInfrastructureTask"
+                failed-async-response:
+                  $ref: "#/components/examples/ExampleFailedInfrastructureTask"
         "400":
           description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+              examples:
+                example:
+                  $ref: "#/components/examples/ExampleCreateInfrastructureErrorResponse"
   /query:
     post:
       tags:
         - infrastructure
-      summary: Query infrastructure create/delete status by ID and deployment location
-      description: Query infrastructure create/delete status by ID and deployment location
+      summary: Query create/delete request status
+      description: >-
+        Query the progress of a create/delete request by ID and deployment location. This API is for debugging as Brent expects the completion of a request to be returned via Kafka
       operationId: .query
       requestBody:
         content:
           application/json:
             schema:
               $ref: "#/components/schemas/InfrastructureTaskQuery"
+            examples:
+              example:
+                $ref: '#/components/examples/ExampleInfrastructureTaskQuery'
       responses:
         "200":
           description: successful operation
           content:
             application/json:
               schema:
                 $ref: "#/components/schemas/InfrastructureTask"
+              examples:
+                complete-example:
+                  $ref: '#/components/examples/ExampleInfrastructureTask'
+                failed-example:
+                  $ref: '#/components/examples/ExampleFailedInfrastructureTask'
         "400":
           description: Invalid ID supplied or infrastructure not found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+              examples:
+                example:
+                  $ref: "#/components/examples/ExampleQueryTaskErrorResponse"
   /delete:
     post:
       tags:
         - infrastructure
       summary: Deletes infrastructure
-      description: Deletes infrastructure
+      description: >-
+        Deletes infrastructure previously created in the given Deployment Location. If the request is accepted, a response should be returned immediately with a requestId to refer to the deletion process. The completion of the request (regardless of whether it was successful or not) should be returned via Kafka
```

Comment:
"should be returned via Kafka" - perhaps indicate that this is handled by Ignition by virtue of the fact that the driver returns a response in its get method?